samples: buck: rework BUCK example into 4 separate ones (#74)
NordicSemiconductor/npmx-zephyr: samples/buck_pins/README.rst

.. _buck_pins_sample:

> BUCK - enable bucks using pin
> #############################

Reviewer: ```suggestion
BUCK - enabling bucks using pin
###############################
```

.. contents::
   :local:
   :depth: 2

The BUCK Pins sample demonstrates how to control PMIC BUCKs using npmx drivers.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM1300 EK.

Overview
********

> This sample uses the npmx BUCK driver to control enable mode for the buck converter with GPIO.
> GPIO3 (:c:enumerator:`NPMX_GPIO_INSTANCE_3`) is set to the input mode and is configured to be used as an enable mode selector for BUCK1 (:c:enumerator:`NPMX_BUCK_INSTANCE_1`).
> The output voltage of BUCK1 is set to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_3V3`).
> When you change the input state of the selected GPIO, the voltage in the BUCK1 output will change from 0 V to the previously selected voltage.

Reviewer 2: ```suggestion
This sample uses the npmx BUCK driver to control the enable mode for the buck converter with GPIO.
GPIO3 (:c:enumerator:`NPMX_GPIO_INSTANCE_3`) is set to the input mode and is configured to be used as an enable mode selector for BUCK1 (:c:enumerator:`NPMX_BUCK_INSTANCE_1`).
The output voltage of BUCK1 is set to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_3V3`).
When you change the input state of the selected GPIO, the voltage in the BUCK1 output will change from 0 V to the previously selected voltage.
```

   .. list-table:: nPM1300 EK connections.
     :widths: 25 25 25 25
     :header-rows: 1

     * - nPM1300 EK pins
       - nRF5340 DK pins
       - nRF52840 DK pins
       - nRF52832 DK pins
     * - GPIO0
       - P1.10
       - P1.10
       - P0.24
     * - SDA
       - P1.02
       - P0.26
       - P0.26
     * - SCL
       - P1.03
       - P0.27
       - P0.27
     * - VOUT2 & GND
       - External supply (P21)
       - External supply (P21)
       - External supply (P21)

#. Make the following connections on the nPM1300 EK:

   * Connect a USB power supply to the **J3** connector.
   * On the **P18** pin header, connect **VOUT2** and **VDDIO** pins with a jumper.
   * On the **P17** pin header, connect all LEDs with jumpers.
   * On the **P13** pin header, connect **VSET1** and **GND** pins with a jumper.
   * On the **P14** pin header, connect **VSET2** and **RSET2** pins with a jumper.

#. Make the following connections on the chosen DK:

   * Set the **SW9** nRF power source switch to **VDD**.
   * Set the **SW10** VEXT → nRF switch to **ON**.

Configuration
*************

|config|

Building and running
********************

> .. |sample path| replace:: :file:`samples/buck`

Reviewer 2: ```suggestion
.. |sample path| replace:: :file:`samples/buck_pins`
```

.. _buck_pins_sample:

BUCK - Enabling bucks using pin
###############################

.. contents::
   :local:
   :depth: 2

> This BUCK sample variant demonstrates how to control PMIC BUCKs with the use of npmx drivers.

Reviewer: ```suggestion
This BUCK sample variant demonstrates how to control the enable mode for the buck converter with GPIO with the use of npmx drivers.
```
